GPflowOpt/scaling.py

```diff
@@ -0,0 +1,229 @@
+        """
+        assert (isinstance(t, DataTransform))
+        Y = self.Y.value
+        self._output_transform.assign(t)
+        self.Y = Y
+
+    @property
+    def normalize_output(self):
+        """
+        :return: boolean, indicating if output is automatically scaled to zero mean and unit variance.
+        """
+        return self._normalize_Y
+
+    @normalize_output.setter
+    def normalize_output(self, flag):
+        """
+        Enable/disable automated output scaling. If switched off, the output transform becomes the identity transform.
+        If enabled, data will be automatically scaled to zero mean and unit variance. When the output normalization is
+        switched on or off, the data in the model is automatically adapted.
+        :param flag: boolean, turn output scaling on or off
+        """
+
+        self._normalize_Y = flag
+        if not flag:
+            self.output_transform = LinearTransform(np.ones(self.Y.value.shape[1]), np.zeros(self.Y.value.shape[1]))
+        else:
+            self.Y = self.Y.value
+
+    # Methods overwriting methods of the wrapped model.
+    @property
+    def X(self):
+        """
+        Returns the input data of the model, unscaled.
+        :return: :class:`.DataHolder`: unscaled input data
+        """
+        return DataHolder(self.input_transform.backward(self.wrapped.X.value))
+
+    @property
+    def Y(self):
+        """
+        Returns the output data of the wrapped model, unscaled.
+        :return: :class:`.DataHolder`: unscaled output data
+        """
+        return DataHolder(self.output_transform.backward(self.wrapped.Y.value))
+
+    @X.setter
+    def X(self, value):
+        """
+        Set the input data. Applies the input transform before setting the data of the wrapped model.
+        """
+        self.wrapped.X = self.input_transform.forward(value)
+
+    @Y.setter
+    def Y(self, value):
+        """
+        Set the output data. In case normalize_Y=True, the appropriate output transform is updated. It is then
+        applied on the data before setting the data of the wrapped model.
+        """
+        if self.normalize_output:
+            self.output_transform.assign(~LinearTransform(value.std(axis=0), value.mean(axis=0)))
+        self.wrapped.Y = self.output_transform.forward(value)
+
+    def build_predict(self, Xnew, full_cov=False):
+        """
+        build_predict builds the TensorFlow graph for prediction. Similar to the method in the wrapped model, however
+        the input points are transformed using the input transform. The returned mean and variance are transformed
+        backward using the output transform.
+        """
+        f, var = self.wrapped.build_predict(self.input_transform.build_forward(Xnew), full_cov=full_cov)
+        return self.output_transform.build_backward(f), self.output_transform.build_backward_variance(var)
+
+    @AutoFlow((float_type, [None, None]))
+    def predict_f(self, Xnew):
```

Comment:
dont we need only to override build_predict? GPFlow Model's predict_f will call our derived one?